contracts/src/token/erc20/extensions/erc4626.rs

```diff
@@ -0,0 +1,528 @@
+//! ERC-4626 Tokenized Vault Standard Implementation
+//!
+//! Extends ERC-20 for vaults, enabling minting and burning of shares
+//! in exchange for underlying assets. This contract provides standardized
+//! workflows for deposits, minting, redemption, and burning of assets.
+//! Note: The shares are minted and burned in relation to the assets via
+//! the `deposit`, `mint`, `redeem`, and `burn` methods, affecting only
+//! the shares token, not the asset token itself.
+//!
+//! [ERC]: https://eips.ethereum.org/EIPS/eip-4626
+
+use alloy_primitives::{Address, U256};
+use alloy_sol_macro::sol;
+use stylus_sdk::{
+    contract, evm, msg,
+    prelude::storage,
+    storage::{StorageU8, TopLevelStorage},
+    stylus_proc::{public, SolidityError},
+};
+
+use crate::{
+    token::erc20::{
+        self,
+        utils::{
+            safe_erc20::{self, ISafeErc20},
+            SafeErc20,
+        },
+        Erc20, IErc20,
+    },
+    utils::math::alloy::{Math, Rounding},
+};
+
+/// ERC-4626 Tokenized Vault Standard Interface
+pub trait IERC4626 {
+    /// Error type associated with the operations, convertible to a vector of
+    /// bytes.
+    type Error: Into<alloc::vec::Vec<u8>>;
+
+    /// Returns the address of the underlying asset that the vault manages.
+    fn asset(&self) -> Address;
```

Comment:
Double check all functions' docs for missing arguments

Reply:
+1